recipe/meta.yaml

```diff
@@ -114,6 +120,8 @@ outputs:
         - cffi  # [py<310]
         - charset-normalizer
         - meson
+        # restrict setuptools: see https://github.com/numpy/numpy/issues/27405
+        - setuptools <74
```

Comment:
Won't be better to temporarily move it to `run_constrained` or provide a [patch](https://github.com/numpy/numpy/pull/27406/files)? Then we'll be confident that the issue won't happen. Someone says that it's a [disaster](https://github.com/numpy/numpy/issues/27405#issuecomment-2378413368) for him.

Reply:
I did add that patch, then had to create yet another patch for additional tests. But that doesn't work for windows where we do want to test the functionalities that use distutils.msvccompiler. 

I will move the restriction to run_constrained with a comment. Note that this does not cover older numpy version, which are affected the same. We can easily add setuptools restriction in feedstocks if we see that issue though.